debian/naivecalendar-update-all-themes.1
.TH NAIVECALENDAR 1 "January 6, 2021"

.SH NAME
naivecalendar-update-all-themes 
- script to update all naivecalendar theme at once


.SH SYNOPSIS
.B naivecalendar-update-all-themes [-h] [-s] [-L] [-p
.I PARAMETER
.B ] [-v 
.I NEW_VALUE
.B ] 

.SH DESCRIPTION
.B naivecalendar-update-all-themes
is a bash script to update a parameter for all naivecalendar themes at once

/!\\ Save your themes before use!

Go to the theme directory and launch

.SH OPTIONS

.TP
.B -h|--help 
show this help message and exit

.TP
.B -s|--simulate 
don't change files and print to stdout

.TP
.B -L|--list-parameters
show all parameters names

.TP
.B -p|--parameter         
parameter to change in all config files

.TP
.B -v|--value             
new value to set



Changelog
v0.7.1 : 

	Minor:
		Capture rofi close error when openning menus
		Save session cmd parameters after openning menu
		events now support strftime format in folder name

v0.7.0 : Again and again and again

	Major:
		Add config option : update message at each date change
		Event : tiny format to parse event for menus
		
	Minor:
		New themes : lead pencil and blackboard
		Update readme : add version number
		Update doc : add example
		Keep whitespace defined in config for alignement control
		Can now set "back to calendar" symbol

v0.6.2 : Improvements

    Minor:
        Improve Completion
        Update Readmes

    Bugs:
        infinite loop with --theme option
        special symbols issues in regex

v0.6.1 : Scripts & Completion

    Major:
        add completion
    Minor:
        improve scripts
    Bugs:
        script now handle special character as space and slash

v0.6.0 : Recursion

    Major:
        Remove unconsistent mode-switcher
        Add script to update all theme at once
        Rewrite menu and shortcut bar construction (prevent unconsitent config)
    Minor:
        Add missing optional dependencies
        Prevent recursive accumulation of rofi calls
    Bugs:
        Prevent crash when wrong theme name is given

v0.5.2 : Themes Conflicts

    Major:
        Now override all theme.rasi files, naivecalendat should appear the same on all configs

v0.5.1 : Help & Readme

    Minor:
        Better help msg
        Update Readme

v0.5.0 : Better header and themes

    Major :
        More configurable header (break old config)
    Minor :
        New themes

V0.4.1 : Alignment

    Minor:
        Align all symbols/character to right

v0.4.0 : Menu & Shortcuts

    Major:
        Add a new bar that can be used for shortcuts and menu
        Reformat cfg files : breaking changes
    Minor:
        Update and add themes variant
        Document config files
    Bugs:
        augment sleep time before relaunch rofi to prevent "lose focus click"
        prevent crash when conflict between old cache file and new config

v0.3.5 : Multiple Events

    Major:
        Add multiple events folder (replace notes)
        Add a menu button
        Better symbols support for event files names
    Minor:
        Update doc / man / readme
    Bugs:
        support name with point for notes (now events)

v0.3.4 : Hello Copy-to-clipboard

    Major:
        add copy-to-clipboard (-x|--clipboard) option
        add version message (-v|--version)
    Minor:
        docstring, manual and message update
        more explicit global var assignement

v0.3.3 : Byebye tkinter

    Major:
        remove tkinter dependancies (rofi popup only)
    Minor:
        document undocumented functions

v0.3.2 : The unamed version

    Major :
        add "open at date" option
    Minor :
        correct numpy docstrings
    Bugs :
        missing dep in deb
        non-recursive config folder creation⎋

v0.3.1 : Debpackage

    Major:
        Create debian package


tools/naivecalendar-update-all-themes.sh
#!/usr/bin/env bash

# Update a naivecalendar theme parameter in all themes, use carefully!

PARAM_LIST="""
USER_LOCALE
PROMT_DATE_FORMAT
IS_TODAY_HEAD_MSG
TODAY_HEAD_MSG_TXT
TODAY_HEAD_MSG_SIZES
TODAY_HEAD_MSG_RISES
DAY_ABBR_LENGHT
FIRST_DAY_WEEK
SYM_NEXT_MONTH
SYM_NEXT_YEAR
SYM_PREV_MONTH
SYM_PREV_YEAR
NB_ROW
ROW_WEEK_SYM
ROW_CAL_START
ROW_CONTROL_MENU
ROW_SHORTCUTS
SYMS_WEEK_DAYS
SYMS_DAYS_NUM
SYMS_CONTROL_MENU
SYMS_SHORTCUTS
SYM_SHOW_HELP
SYM_SWITCH_THEME
SYM_SHOW_EVENTS
SYM_SWITCH_EVENT
SYM_SHOW_MENU"""

help_msg () {
    echo """
usage: naivecalendar-update-all-themes.sh [-h] [-L] [-f FILE] [-p PARAMETER] [-v NEW_VALUE]

Give a new value to a calendar parameter for all themes. /!\ Save your themes before use!

optional arguments:
  -h, --help              show this help message and exit
  -s, --simulate          don't change files and print to stdout
  -L, --list-parameters   show all parameters names
  -p, --parameter         parameter to change in all config files
  -v, --value             new value to set
"""
}

list_param () {
    printf "naivecalendar theme parameters: \n\n"
    printf "%-25s  %-25s  %-25s  %-25s\n"$PARAM_LIST
}


# Save parameters
option="$@"

# Loop over named parameters
while [ $# -gt 0 ] ; do
    key="$1"
    case $key in
        -h|--help)
            help_msg 
            exit 0
            ;;
        -p|--parameter)
            PARAM="$2"
            ;;
        -s|--simulate)
            ;;
        -v|--value)
            VALUE="$2"
            ;;
        -L|--list-parameters)
            list_param
            exit 0
            ;;
        *)
          printf "**************************\n"
          printf " Wrong argument : $1\n"
          printf "**************************\n"
          exit 1
    esac
    shift # past argument
    shift # past value
done

# Test parameters validity
if [ -z "$PARAM" ]
  then
    printf "***********************************************\n"
    echo "Please give a parameter to set, see -L option"
    printf "***********************************************\n"
    exit 0
fi

if echo $PARAM_LIST | grep -w $PARAM > /dev/null; then
    printf "setting parameter $PARAM "
else
    printf "*********************************************************\n"
    printf "$PARAM in not in the parameter list, see -L option \n"
    printf "*********************************************************\n"
    exit 0
fi

if [ -z "$VALUE" ]
  then
    printf "\n***********************************************\n"
    printf "Please give a value for $PARAM \n"
    printf "***********************************************\n"
    exit 0
else
    printf "to $VALUE \n"
fi



# Replace the parameter value or just print to stdout if simulate
if [[ " ${option[@]} " =~ " -s " ]] || [[ " ${option[@]} " =~ " --simulate " ]] ; then
    for f in *.cfg; do
        sed -r 's/('$PARAM' =).*/\1 '$VALUE'/g' $f
    done
    printf "remove '-s|--simulate' to apply changes\n"
else
    i=0
    for f in *.cfg; do
        sed -i -r 's/('$PARAM' =).*/\1 '$VALUE'/g' $f
        i=$(($i+1))
    done
    printf "$i files updated\nBye!\n"
fi

